src/components/ebay-pagination/index.js

```diff
@@ -0,0 +1,173 @@
+    const inputItems = input.items || [];
+    const isRoleButton = input.fakeLink || false;
+    let role;
+
+    if (isRoleButton) {
+        role = 'button';
+    }
+
+    for (let i = 0; i < inputItems.length; ++i) {
+        const item = inputItems[i];
+        const href = item.href;
+        const current = item.current;
+
+        tempObj = {
+            role,
+            href,
+            current: Boolean(current) || false,
+            htmlAttributes: processHtmlAttributes(item),
+            renderBody: item.renderBody
+        };
+        if (item.previous) {
+            prevItem = tempObj;
+            prevItem.class = 'pagination__previous';
+            prevItem.disabled = Boolean(item.disabled) || !href;
+            continue;
+        }
+        if (item.next) {
+            nextItem = tempObj;
+            nextItem.class = 'pagination__next';
+            nextItem.disabled = Boolean(item.disabled) || !href;
+            continue;
+        }
+        items.push(tempObj);
+    }
+    if (!prevItem) {
+        prevItem = {
+            class: 'pagination__previous',
+            disabled: true
+        };
+    }
+
+    if (!nextItem) {
+        nextItem = {
+            class: 'pagination__next',
+            disabled: true
+        };
+    }
+
+    return {
+        isRoleButton,
+        nextItem,
+        prevItem,
+        items,
+        ariaLabelPrev: input.ariaLabelPrev || 'Previous page',
+        ariaLabelNext: input.ariaLabelNext || 'Next page',
+        currText: input.currText || 'Results Pagination - Page 1',
+        classes: ['pagination', input.class],
+        htmlAttributes: processHtmlAttributes(input)
+    };
+}
+
+function getTemplateData(state) {
+    return state;
+}
+
+function init() {
+    this.pageContainerEl = this.el.querySelector('.pagination__items');
+    this.pageEls = this.pageContainerEl.children;
+    this.containerEl = this.el;
+    this.subscribeTo(eventUtils.resizeObserver).on('resize', refresh.bind(this));
+    this.refresh();
+}
+
+function refresh() {
+    const containerWidth = this.containerEl.offsetWidth;
+    const pageNumWidth = this.pageEls[0].offsetWidth + constants.margin;
+    const numPagesAllowed = Math.floor(containerWidth / pageNumWidth) - constants.indexForNavigation;
+    const adjustedNumPages = Math.min(constants.maxPagesAllowed,
+        Math.max(numPagesAllowed, constants.minPagesRequired));
+    const totalPages = this.pageEls.length;
+
+    // Let's show all the pages that we can.
+    for (let i = 5; i < adjustedNumPages && i < totalPages; ++i) {
+        if (this.pageEls[i].hasAttribute('hidden')) {
+            this.pageEls[i].removeAttribute('hidden');
+        }
+    }
+
+    // Now that we are showing all the pages that we can, lets hide remaining pages.
+    for (let i = adjustedNumPages; i < totalPages; ++i) {
+        this.pageEls[i].setAttribute('hidden', true);
+    }
+}
+
+function isPreventDefault(e) {
+    if (this.state.isRoleButton) {
```

Comment:
`isRoleButton` makes sense above, but not here. Maybe rename it to `hasFakeLinks`